src/PSRule/Runtime/AssertResult.cs

```diff
@@ -124,7 +124,7 @@ public AssertResult ReasonIf(bool condition, string text, params object[] args)
         /// <returns>Returns an array of reasons. This will always return null when the Value is true.</returns>
         public string[] GetReason()
         {
-            if (!Result || _Reason == null || _Reason.Count == 0)
+            if (Result || _Reason == null || _Reason.Count == 0)
```

Comment:
Had to change this to return an empty array if the result is `true`. Let me know if this is incorrect or I've missed something.

Reply:
@ArmaanMcleod Yes this looks like a bug. Thanks for fixing it.